File: CS/WinSolution.Module.Win/Updater.cs
using System;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp.Reports;
using DevExpress.ExpressApp.Updating;

namespace WinSolution.Module.Win {
    public class Updater : ModuleUpdater {
        public Updater(Session session, Version currentDBVersion) : base(session, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            string reportName = "TestReport";
            string[] resources = GetType().Assembly.GetManifestResourceNames();
            string reportResourceName = Array.Find<string>(resources, IsTestReport);
            ReportData reportdata = Session.FindObject<ReportData>(new BinaryOperator("Name", reportName));
            if (reportdata == null && !string.IsNullOrEmpty(reportResourceName)) {
                reportdata = new ReportData(Session);
                XafReport rep = new XafReport();
                rep.ReportName = reportName;
                rep.ObjectSpace = new ObjectSpace(new UnitOfWork(Session.DataLayer), XafTypesInfo.Instance);
                rep.LoadLayout(GetType().Assembly.GetManifestResourceStream(reportResourceName));
                reportdata.SaveXtraReport(rep);
                reportdata.Save();
            }
        }
        private bool IsTestReport(string reportName) {
            return reportName.EndsWith("TestReport.repx");
        }
    }
}

File: VB/WinSolution.Module.Win/Updater.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp.Reports
Imports DevExpress.ExpressApp.Updating

Namespace WinSolution.Module.Win
    Public Class Updater
        Inherits ModuleUpdater
        Public Sub New(ByVal session As Session, ByVal currentDBVersion As Version)
            MyBase.New(session, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            Dim reportName As String = "TestReport"
            Dim resources() As String = Me.GetType().Assembly.GetManifestResourceNames()
            Dim reportResourceName As String = Array.Find(Of String)(resources, AddressOf IsTestReport)
            Dim reportdata As ReportData = Session.FindObject(Of ReportData)(New BinaryOperator("Name", reportName))
            If reportdata Is Nothing AndAlso (Not String.IsNullOrEmpty(reportResourceName)) Then
                reportdata = New ReportData(Session)
                Dim rep As New XafReport()
                rep.ReportName = reportName
                rep.ObjectSpace = New ObjectSpace(New UnitOfWork(Session.DataLayer), XafTypesInfo.Instance)
                rep.LoadLayout(Me.GetType().Assembly.GetManifestResourceStream(reportResourceName))
                reportdata.SaveXtraReport(rep)
                reportdata.Save()
            End If
        End Sub
        Private Function IsTestReport(ByVal reportName As String) As Boolean
            Return reportName.EndsWith("TestReport.repx")
        End Function
    End Class
End Namespace